### PR TITLE
Added timestamp conversion utlities. I was tired of all the error-pro…

### PR DESCRIPTION
**Pull request by JonathanHenson**
…ne duplicate timestamp conversions.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/windows/clock.c [hunk 1]
```diff
@@ -59,8 +57,8 @@ int aws_high_res_clock_get_ticks(uint64_t *timestamp) {
     LARGE_INTEGER ticks, frequency;
     /* QPC runs on sub-microsecond precision, convert to nanoseconds */
     if (QueryPerformanceFrequency(&frequency) && QueryPerformanceCounter(&ticks)) {
```
Comment (graebm) [inline source/windows/clock.c]: QueryPerformanceFrequency() returns the same result for the life of the program, see [docs](https://msdn.microsoft.com/en-us/library/windows/desktop/ms644904(v=vs.85).aspx). You could save the value in a static int.

File: source/windows/clock.c [hunk 2]
```diff
@@ -59,8 +57,8 @@ int aws_high_res_clock_get_ticks(uint64_t *timestamp) {
...
-
-        *timestamp = (((uint64_t)ticks.QuadPart * MUS_PER_SEC) / (uint64_t)frequency.QuadPart) * NS_PER_MUS;
+        *timestamp = aws_timestamp_convert((uint64_t)ticks.QuadPart, AWS_TIMESTAMP_SECS, AWS_TIMESTAMP_MICROS);
+        *timestamp /= aws_timestamp_convert((uint64_t)frequency.QuadPart, AWS_TIMESTAMP_MICROS, AWS_TIMESTAMP_NANOS);
         return AWS_OP_SUCCESS;
```
[the remark this answers is not in the document]
Comment (JonathanHenson) [reply source/windows/clock.c]: i don't like any of it, and it's a shitty api on MSFT's part. I don't care which poison we pick, but I'm partial to the one that doesn't involve a change unless you really care.

File: source/windows/clock.c (hunk 1 above)
Comment (JonathanHenson) [reply source/windows/clock.c]: It's static under the hood anyways. and it's not really related to this change.

File: source/posix/condition_variable.c [hunk 3]
```diff
@@ -88,8 +86,8 @@ int aws_condition_variable_wait_for(
     time_to_wait += current_sys_time;
 
     struct timespec ts;
-    ts.tv_sec = time_to_wait / NANOS_PER_SEC;
-    ts.tv_nsec = time_to_wait % NANOS_PER_SEC;
+    ts.tv_sec = aws_timestamp_convert((uint64_t)time_to_wait, AWS_TIMESTAMP_NANOS, AWS_TIMESTAMP_SECS);
+    ts.tv_nsec = time_to_wait % AWS_TIMESTAMP_NANOS;
```
[the remark this answers is not in the document]
Comment (JonathanHenson) [reply source/posix/condition_variable.c]: Then I'd have to calculate it everywhere (mod isn't cheap), when we really only need it in common, and this is implemented for everywhere else in the code base that just needs to do some basic clock conversions.

[the remark this answers is not in the document]
Comment (JonathanHenson) [reply source/posix/condition_variable.c]: I will do this one.

**Pull request merged**